terraform/modules/single_instance/cron.tf

```diff
@@ -0,0 +1,71 @@
+variable "crontabs" {
+  type = map(object({
+    cron_time = string
+    cron_user = optional(string,"root")
+    script = string 
+  }))
+  default = {}
+  validation {
+    condition = !( length(var.crontabs) > 0 && var.is_windows )
+    error_message = "Can't add crons to windows!"
+  }
+}
+
+resource "null_resource" "cron" {
+  for_each = {
+    for k,v in local.crons : k => v
+    if local.scripts_enabled
+  }
+  depends_on = [ null_resource.testconnection ]
+  triggers = {
+    user = local.ssh_user
+    port = local.ports.ssh
+    ip = data.openstack_networking_floatingip_v2.public.address
+    content = each.value.script
+  }
+  connection {
+    type     = "ssh"
+    user     = self.triggers.user
+    agent = true
+    host     = self.triggers.ip
+    timeout = "5m"
+    port = self.triggers.port
+  }
+  provisioner "file" {
+    destination = "/home/${local.ssh_user}/${random_id.obscure[each.key].id}-${each.key}.sh"
+    content = each.value.script
+  }
+  provisioner "file" {
+    destination = "/home/${local.ssh_user}/${random_id.obscure[each.key].id}-${each.key}.cron"
+    content = <<-EOT
+    ${each.value.cron_time} ${each.value.cron_user} /opt/vsc/cron/${each.key}.sh
+
+    EOT
+  }
+  provisioner "remote-exec" {
```

Comment:
not local-exec?

Reply:
I do have a Proof Of Concept for an ansible playbook. Only downside is that it doesn't really support the "time format string" like `* * * * *` and instead needs dedicated variables for minute/hour/day/weekday/...